common/lib/capa/capa/inputtypes.py

```diff
@@ -232,7 +234,7 @@ def __init__(self, system, xml, state):
 
         # put hint above msg if it should be displayed
         if self.hintmode == 'always':
-            self.msg = self.hint + ('<br/>' if self.msg else '') + self.msg
+            self.msg = HTML('{hint}<br/>{msg}' if self.msg else '').format(hint=self.hint, msg=self.msg)
```

Comment:
@staubina it is my understanding the default for input args (for both HTML and Text methods) is Text. For it not to be escaped, self.hint and self.msg need to be wrapped with HTML. In which case you might be best just going back to concatenation....

Note also that what you have here is not equivalent to the original. You have dropped hints entirely when self.msg is empty.